test/src/test/java/hudson/model/DescriptorTest.java

```diff
@@ -68,14 +68,14 @@
     @Test public void overriddenId() throws Exception {
         FreeStyleProject p = rule.createFreeStyleProject();
         p.getBuildersList().add(new BuilderImpl("builder-a"));
-        rule.configRoundtrip(p);
+        rule.configRoundtrip((Item)p);
```

Comment:
Not eclipse specific - the JLS is a little ambiguous in this area, eclipse compiler happened to choose one way - the JDK chose another.  Arguably I think that eclipse is the better approach.
now that test harness has gone from core, I can probably revisit #1830